persistent-mysql/Database/Persist/MySQL.hs

```diff
@@ -500,7 +500,7 @@ findTypeOfColumn allDefs name col =
         ((,) col)
         $ do
             entDef   <- find ((== name) . getEntityDBName) allDefs
-            fieldDef <- find ((== col)  . fieldDB) (getEntityFields entDef)
+            fieldDef <- find ((== col)  . fieldDB) (getEntityFieldsDatabase entDef)
```

Comment:
Lots of this is in the backend modules. We need *all* the columns, not just the Haskell fields.